transports/sqs/configuration-options_visibility-timeout_sqs_[7.3,).partial.md

```diff
@@ -0,0 +1,87 @@
+
+```text
+buffer = Min(remainingTime / 2, 10 seconds)
+renewAfter = remainingTime - buffer
+```
+
+Renewal is triggered after `renewAfter` has elapsed since the last renewal. If the remaining time is less than 400ms, the renewal is attempted immediately. This ensures the message visibility is extended consistently and early enough to avoid expiration, even under system load or scheduling delays.
+
+When renewing, the transport extends the visibility timeout by:
+
+```text
+Max(Abs(remainingTimeInSeconds) + visibilityTimeoutInSeconds, visibilityTimeoutInSeconds)
+```
+
+This allows the visibility to be “pushed forward” by a consistent chunk while also compensating for delays. If a renewal is delayed and occurs after the visibility timeout expires, the transport tries to compensate, for example assuming the remaining time is `-2s` the new visibility timeout would be `abs(-2s) + 30s = 32s`. This is done because an expired visibility timeout doesn’t always mean the message can’t be completed—if no competing consumer picks it up, processing may still succeed.
+
+**Example**:
+
+- Configured visibility timeout: 30 seconds
+- Max total extension duration: 5 minutes
+
+```mermaid
+sequenceDiagram
+    participant Queue
+    participant Handler
+
+    Note over Queue,Handler: T+0s: Message received
+    Queue->>Handler: Deliver message
+    Note right of Handler: Initial visibility = 30s (expires at T+30s)
+
+    Note over Handler: Processing...
+
+    Note over Handler: Schedule renewal at T+20s
+    Handler->>Queue: T+20s: Renew (remaining = 10s)
+    Note right of Queue: Extend visibility to T+60s (extended by 10s + 30s = 40s)
+
+    Note over Handler: Schedule next renewal at T+50s
+    Handler->>Queue: T+50s: Renew (remaining = 10s)
+    Note right of Queue: Extend visibility to T+90s (extended by 10s + 30s = 40s)
+
+    Note over Handler: Schedule next renewal at T+80s
+    Handler->>Queue: T+80s: Renew (remaining = 10s)
+    Note right of Queue: Extend visibility to T+120s (extended by 10s + 30s = 40s)
+
+    alt Delayed renewal (remaining < 0)
+        Handler->>Queue: Overextend visibility by abs(remaining) + 30s
+    end
+
+    alt T+300s reached
+        Note right of Handler: Stop renewing
+    end
+
+    alt Processing completes
+        Handler->>Queue: Delete message
+    else Handler still running
+        Queue->>Handler: Message becomes visible again
+    end
+```
+
+The message visibility timeout extension is a best effort operation and can fail due to several reasons including:
+
+- Larger clock skew between the client and the SQS service time
+- Network interruptions for longer periods of time between the client and the SQS service
+- Backpressure from the SQS service due to throttling
+
+### MaxAutoMessageVisibilityRenewalDuration
+
+**Optional**
+
+**Default**: `TimeSpan.FromMinutes(5)`
+
+This allows increasing or decreasing the maximimum time the message visibility renewal is attempted during message processing.
```

Comment:
The way I read this sentence is that the transport tries to renew for five minutes and then stops. How about the following:

```suggestion
This configures the allowed maximum message visibility timeout, after which the transport stops renewing the visibility. Amazon SQS supports a maximum value of 12 hours.
```